### ISSUE TITLE
Vue 3 version

### ISSUE THREAD

**Issue by AlexandreBonaventure** [posva/focus-trap-vue#226]
Hello 👋 
For my own project I forked and _upgraded_ this library with Vue 3 API. Would you be interested in joining our effort to merge these changes for a vue 3 compat ?

looks roughly like that 
````ts
import { defineComponent, onMounted, watch, ref, cloneVNode, onUnmounted } from 'vue';
import createFocusTrap, { FocusTrap as FocusTrapI } from 'focus-trap';

const FocusTrap = defineComponent({
  props: {
    active: {
      // TODO: could be options for activate
      type: Boolean as () => boolean,
      default: true,
    },
    escapeDeactivates: {
      type: Boolean as () => boolean,
      default: true,
    },
    returnFocusOnDeactivate: {
      type: Boolean as () => boolean,
      default: true,
    },
    allowOutsideClick: {
      type: Boolean as () => boolean,
      default: true,
    },
    initialFocus: {
      type: [String as () => string, Function as () => () => HTMLElement],
      default: undefined,
    },
    fallbackFocus: {
      type: [String as () => string, Function as () => () => HTMLElement],
      default: undefined,
    },
  },
  setup (props, { slots, emit }) {
    let trap: FocusTrapI | null;
    const el = ref<HTMLElement | null>(null);
    onMounted(function () {
      watch(() => props.active, (active) => {
        if (active && el.value) {
          // has no effect if already activated
          trap = createFocusTrap(
            el.value,
            {
              escapeDeactivates: props.escapeDeactivates,
              allowOutsideClick: () => props.allowOutsideClick,
              returnFocusOnDeactivate: props.returnFocusOnDeactivate,
              onActivate: () => {
                emit('update:active', true);
                emit('activate');
              },
              onDeactivate: () => {
                emit('update:active', false);
                emit('deactivate');
              },
              initialFocus: typeof props.initialFocus === 'string' ? props.initialFocus : props.initialFocus?.() ?? el.value,
              fallbackFocus: props.fallbackFocus,
            },
          );
          trap.activate();
        } else {
          trap?.deactivate();
        }
      },
      { immediate: true },
      );
    });
    onUnmounted(() => {
      trap?.deactivate();
      trap = null;
    });
    return () => {
      const content = slots.default?.();
      if (!content || !content.length || content.length > 1) { throw new Error('FocusTrap requires exactly one child'); };
      const vnode = cloneVNode(content[0], { ref: el });
      return vnode;
    };
  },
});

export { FocusTrap };

```

**Comment by posva** [comment]
Hi, sure, we can put the work on a branch and release it under a `next` tag

**Comment by posva** [comment]
I forgot to say, I invited you as a collaborator to the project!

**Comment by AlexandreBonaventure** [comment]
Thanks, I started a WIP PR not ready for prime time. I'll update when I find some free time. Let me know if you have any concern (Tu parles français aussi n'est-ce pas :) ?)

**Comment by posva** [comment]
Cool thanks! Oui, je parle français, si une discussion a plus de sense d'être en français, n'hésites pas. De mon côté je ferais les reviews et tout en anglais :)

**Comment by AlexandreBonaventure** [comment]
released in v2.0.0
